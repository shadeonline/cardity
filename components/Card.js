import React from 'react';
import { useNavigation } from '@react-navigation/native';
import { Text, View, Image, TouchableOpacity } from 'react-native';
import { imageMap } from '../imageMap';
import { Cell } from 'react-native-tableview-simple';

const Card = ({ cardName, cardId, redirect, card, image, color }) => {
  const navigation = useNavigation();
  // console.log(cardName, cardId, redirect, card, image, color);

  // Check if image is undefined and use color as background
  const imageSource = image ? imageMap[image] : null;
  const cardColor = image ? 'transparent' : color;

  // Check if redirect is blank or not. If redirect is not blank, then navigate to the desired view, else do nothing
  const handlePress = () => {
    if (redirect === 'Details') {
      navigation.navigate('Details', { card });
    }
  };

  // Conditionally render the image based on whether the image property is defined
  const imageContent = image ? (
    <Image source={imageSource} style={styles.image} resizeMode="cover" />
  ) : null;

  const cardContent = redirect ? (
    <Cell
      contentContainerStyle={styles.containerStyle}
      cellContentView={
        <TouchableOpacity onPress={handlePress} style={[styles.cardContainer, { backgroundColor: cardColor }]}>
          <View style={styles.cardContent}>
            {imageContent}
            <Text style={styles.cardNameText}>{cardName}</Text>
            <Text style={styles.cardIdText}>Card No.: {cardId}</Text>
          </View>
        </TouchableOpacity>
      }
    />
  ) : (
    <Cell
      contentContainerStyle={styles.containerStyle}
      cellContentView={
        <View onPress={handlePress} style={[styles.cardContainer, { backgroundColor: cardColor }]}>
          <View style={styles.cardContent}>
            {imageContent}
            <Text style={styles.cardNameText}>{cardName}</Text>
            <Text style={styles.cardIdText}>Card No.: {cardId}</Text>
          </View>
        </View>
      }
    />
  );

  return cardContent;
};


const styles = {
  containerStyle: {
    alignItems: 'center',
    justifyContent: 'center',
    height: 250,
    width: '100%',
    backgroundColor: 'transparent',
    highlightColor: '#ccc',
    padding: 20,
    marginVertical: 40,
  },
  image: {
    width: '100%', // Use '100%' to ensure the image takes the full width
    // aspectRatio: 16 / 9, // Set the aspect ratio to maintain the image's original aspect ratio (e.g., 16:9)
    height:'100%',
    padding: 0,
    margin: 0,
    position: 'absolute',
  },
  cardContainer: {
    borderRadius: 10,
    width: '100%',
    height: 260, // Set a fixed height here to match the container size
    overflow: 'hidden', // Hide any content that overflows the card
  },
  cardContent: {
    flex: 1,
    alignItems: 'center',
    justifyContent: 'center',
  },
  cardNameText: {
    position: 'absolute',
    top: '20%',
    left: '5%',
    fontSize: 24,
    backgroundColor: 'white',
    padding: 5,
  },
  cardIdText: {
    position: 'absolute',
    top: '75%',
    left: '5%',
    fontSize: 15,
    backgroundColor: 'white',
    padding: 5,
  },
};


export default Card;
