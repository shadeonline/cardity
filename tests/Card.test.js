import React from 'react';
import { render } from 'react-native-testing-library';
import Card from '../components/Card'; // Import your Card component

// Mock the useNavigation hook
jest.mock('@react-navigation/native', () => ({
  useNavigation: () => ({
    navigate: jest.fn(),
  }),
}));

describe('Card Component', () => {
  it('matches snapshot', () => {
    const props = {
      cardName: 'The Ice Cream Card',
      cardId: '992847266036',
      redirect: undefined,
      card: {
        "cardId": 992847266036,
        "cardName": "The Ice Cream Card",
        "color": undefined,
        "image": "icecreamRestaurant",
        "loyaltyCard": "loyaltyCards/6xHLagIWsk8Z8Z4ndbO4",
        "loyaltyProgram": "loyaltyPrograms/kAdCUvzu63ybzK09VXEI",
        "progress": 5
      }, // Provide relevant card data here
      image: 'icecreamRestaurant',
      color: undefined,
    };

    const { toJSON } = render(<Card {...props} />);

    // Use Jest's expect to match the snapshot
    expect(toJSON()).toMatchSnapshot();
  });
});
