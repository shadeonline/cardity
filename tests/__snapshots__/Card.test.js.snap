// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component matches snapshot 1`] = `
<View>
  <RCTSafeAreaView
    style={
      [
        {
          "backgroundColor": "#FFF",
        },
        {
          "flexGrow": 1,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "alignItems": "center",
            "flexDirection": "row",
            "minHeight": 44,
            "paddingLeft": 15,
            "paddingRight": 20,
          },
          {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "height": 250,
            "highlightColor": "#ccc",
            "justifyContent": "center",
            "marginVertical": 40,
            "padding": 20,
            "width": "100%",
          },
        ]
      }
    >
      <View
        onPress={[Function]}
        style={
          [
            {
              "borderRadius": 10,
              "height": 260,
              "overflow": "hidden",
              "width": "100%",
            },
            {
              "backgroundColor": "transparent",
            },
          ]
        }
      >
        <View
          style={
            {
              "alignItems": "center",
              "flex": 1,
              "justifyContent": "center",
            }
          }
        >
          <Image
            resizeMode="cover"
            source={1}
            style={
              {
                "height": "100%",
                "margin": 0,
                "padding": 0,
                "position": "absolute",
                "width": "100%",
              }
            }
          />
          <Text
            style={
              {
                "backgroundColor": "white",
                "fontSize": 24,
                "left": "5%",
                "padding": 5,
                "position": "absolute",
                "top": "20%",
              }
            }
          >
            The Ice Cream Card
          </Text>
          <Text
            style={
              {
                "backgroundColor": "white",
                "fontSize": 15,
                "left": "5%",
                "padding": 5,
                "position": "absolute",
                "top": "75%",
              }
            }
          >
            Card No.: 
            992847266036
          </Text>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</View>
`;
